admin/src/components/Users.tsx

```diff
@@ -163,6 +200,50 @@ function getColumns(setRowsData: any, serverData: any) {
           >
             {EDIT_SVG}
           </button>
+          <button
+            title="Block/Unblock"
+            style={styles.cancelButton}
+            onClick={async (e) => {
+              e.stopPropagation();
+
+              // Toggle isBanned status
+              const newStatus = !data.isBanned;
+              await serverData.banUser(data.id, newStatus);
+              data.isBanned = newStatus;
+
+              setRowsData((prevRows: any[]) =>
+                prevRows.map((row) =>
+                  row.id === data.id ? { ...row, isBanned: newStatus } : row
+                )
+              );
+              alert(
+                `User ${data.username} has been ${
+                  newStatus ? "blocked" : "unblocked"
+                }.`
+              );
+            }}
+          >
+            {BLOCK_SVG}
+          </button>
+          <button
+            title="Delete"
+            style={styles.cancelButton}
+            onClick={(e) => {
+              e.stopPropagation();
+
+              if (
+                window.confirm("Are you sure you want to delete this user?")
+              ) {
+                setRowsData((prevRows: any[]) =>
```

Comment:
Amazing job implementing the blocking feature! One thing I would consider is using a map for the row data (such as prevRows) instead of an array to optimize updates for larger datasets. This could allow for constant-time lookups and updates.